packages/core/src/plugins/minimize.ts

```diff
@@ -99,22 +63,17 @@ export const pluginMinimize = (): RsbuildPlugin => ({
 
     api.modifyBundlerChain(async (chain, { isProd, environment, CHAIN_ID }) => {
       const { config } = environment;
-      const isMinimize = isProd && config.output.minify !== false;
-
-      if (!isMinimize) {
-        return;
-      }
-
-      const { SwcJsMinimizerRspackPlugin, LightningCssMinimizerRspackPlugin } =
-        rspack;
+      const { minifyJs, minifyCss, jsOptions, cssOptions } = parseMinifyOptions(
+        config,
+        isProd,
+      );
 
-      const { minifyJs, minifyCss, jsOptions, cssOptions } =
-        parseMinifyOptions(config);
+      chain.optimization.minimize(minifyJs || minifyCss);
```

Comment:
`chain.optimization.minimize(minifyJs || minifyCss);` is set by the basic plugin, should we set it here again?

Reply:
> `chain.optimization.minimize(minifyJs || minifyCss);` is set by the basic plugin, should we set it here again?

it is for a such edge case that `pluginBasic` is not applied - typically seen in test cases.

actually i'm not sure why it's in `pluginBasic`

Reply:
Yeah we can also remove the `chain.optimization.minimize()` from basic plugin.

But this SWC/LightningCSS minimizer plugin is skipped in the webpack mode 😂, so we need to set ``chain.optimization.minimize()` before this `if` condition.

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/0e481b24-12d2-4aa8-b580-dd820f2c2b66)

Reply:
i've refined this PR, please check again.